octodns/provider/route53.py

```diff
@@ -29,19 +31,84 @@ def _octal_replace(s):
 class _Route53Record(object):
 
     @classmethod
-    def new(self, provider, record, creating):
+    def _new_dynamic(cls, provider, record, hosted_zone_id, creating):
         ret = set()
-        if getattr(record, 'geo', False):
-            ret.add(_Route53GeoDefault(provider, record, creating))
-            for ident, geo in record.geo.items():
-                ret.add(_Route53GeoRecord(provider, record, ident, geo,
-                                          creating))
-        else:
-            ret.add(_Route53Record(provider, record, creating))
+
+        # HostedZoneId wants just the last bit, but the place we're getting
+        # this from looks like /hostedzone/Z424CArX3BB224
+        hosted_zone_id = hosted_zone_id.split('/', 2)[-1]
+
+        # Create the default pool
+        fqdn = record.fqdn
+        ret.add(_Route53Record(provider, record, creating,
+                               '_octodns-default-pool.{}'.format(fqdn)))
+
+        # Pools
+        for pool_name, pool in record.dynamic.pools.items():
+
+            # Create the primary
+            ret.add(_Route53DynamicPool(provider, hosted_zone_id, record,
+                                        pool_name, creating))
+
+            # Create the fallback
+            fallback = pool.data.get('fallback', False)
+            if fallback:
+                # We have an explicitly configured fallback
+                ret.add(_Route53DynamicPool(provider, hosted_zone_id, record,
+                                            pool_name, creating,
+                                            target_name=fallback))
+            else:
+                # We fallback on the default
+                ret.add(_Route53DynamicPool(provider, hosted_zone_id, record,
+                                            pool_name, creating,
+                                            target_name='default'))
+
+            # Create the values
+            for i, value in enumerate(pool.data['values']):
+                weight = value['weight']
+                value = value['value']
+                ret.add(_Route53DynamicValue(provider, record, pool_name,
+                                             value, weight, i, creating))
+
+        # Rules
+        for i, rule in enumerate(record.dynamic.rules):
+            pool_name = rule.data['pool']
+            geos = rule.data.get('geos', [])
+            if geos:
+                for geo in geos:
+                    ret.add(_Route53DynamicRule(provider, hosted_zone_id,
+                                                record, pool_name, i,
+                                                creating, geo=geo))
+            else:
+                ret.add(_Route53DynamicRule(provider, hosted_zone_id, record,
+                                            pool_name, i, creating))
```

Comment:
We have no geo configured so we're take anything that's left and point it to the configured pool. So long as it's healthy it'll pick up all the otherwise untargeted stuff.